source/module_base/gram_schmidt_orth-inl.h

```diff
@@ -43,15 +43,21 @@ std::vector<std::vector<Func_Type>> Gram_Schmidt_Orth<Func_Type,R_Type>::cal_ort
 	
 	for( size_t if1=0; if1!=func.size(); ++if1 )
 	{
+		//use CGS2 algorithm to do twice orthogonalization 
+		//DOI 10.1007/s00211-005-0615-4
 		std::vector<Func_Type> func_try = func[if1];
-		for( size_t if_minus=0; if_minus!=func_new.size(); ++if_minus )
+		for(int niter=0;niter<3;niter++)
```

Comment:
What's the point of setting `niter` to 3? I remember this number being 2 when I reviewed.

Reply:
@pxlxingliang Will you look into it?

Reply:
niter = 2 means the CGS2 algorithm, but it will still cause the fail of orthogonalization (running on git hub) when the basis number is 50, so upgrade to CGS3 algorithm by set niter = 3.